### DISCUSSION TITLE
RPC client without chaining, especially with Zod OpenAPI

### DISCUSSION THREAD

**Discussion by fend25** [honojs/hono#3266] [Q&A]
Hi everyone!

First of all, thank you, @yusukebe, for the great work you've done!

I have a question: Is there any option to build an RPC client (using AppType from the docs) without chaining all the routes together?

The documentation suggests that something like this is possible:

```ts
const app1 = new Hono()
app1.route('/1', () => {})

const app2 = new Hono()
app2.route('/2', () => {})

const routes = app.route('/app1', app1).route('/app2', app2)
const client = hc<typeof routes>()
```

However, there's a significant issue: Is it possible to access something like `typeof routes` without relying on extensive global method chaining? I'm thinking of something like `typeof app.routes` or a similar alternative.

Apart from the inconvenience of chaining all methods, there's an issue with Zod-OpenAPI. While Zod-OpenAPI works wonderfully (again, thank you, Wada-sensei!), the problem arises because `app.openapi` doesn't accept parameters like `app.route`.   
Using `app.route('/prefix', new OpenAPIHono())` results in losing route information, leaving the client empty and the IDE indicating that `typeof routes` is also empty.

Given that chaining all the app routes together — especially when using OpenAPI (which adds complexity) — is cumbersome, how should I handle this situation?

And is there a way to call typeof routes without chaining all the server's handlers?

**Comment by Liamandrew** [comment]
I'd like to add to this, I've experienced significant IDE performance issues from chaining everything together. To the point where it was unusable and is a big blocker for me.

I've followed the different guides about app splitting but I ultimately found if everything is chained together then there will still be a huge IDE performance cost of TS trying to infer all the types.

I define the Zod Open API routes separately in my monorepo with `@hono/zod-openapi`, so I'm thinking there must surely be a way to generate a type safe client from this static configuration in an efficient way 🤔
